SpeedrunTimer/data/tables/speedrun-timer-sct.tbm

```diff
@@ -0,0 +1,149 @@
+        self.LSEndpoint = "\\\\.\\pipe\\LiveSplit" --Localhost LiveSplit pipe. Will not likely work with a file.
+        self.InputOutput = require "io" --io is defaulted to the controls. Since we need it to send LiveSplit commands, we need an alias.
+        self.LSPipe = self.InputOutput.open(self.LSEndpoint, "w") --Open/start the pipe. Flush is required after every command.
+        ba.println("Speedrun Timer: Initialized - " .. tostring(time.getCurrentTime()))
+        self.Initialized = true
+    end
+
+    function SpeedrunTimer:StopTimer()
+        self.RunningStatus = false
+        ba.println("Speedrun Timer: StopTimer called - Last Segment Time: " .. self.CurrentSegmentTime)
+        self.CurrentSegmentTime = 0
+        self:PauseLiveSplit();
+    end
+
+    function SpeedrunTimer:PauseLiveSplit() --Send Pause to LiveSplit, and split if its on a tracked segment.
+        ba.println("Speedrun Timer: PauseLiveSplit called - Mission Filename: " .. mn.getMissionFilename())
+        if not(mn.isTraining() or --Training Missions
+         mn.getMissionFilename():upper() == "SM1-01" or --Start of the game.
+         (self.CurrentMission == mn.getMissionFilename() and --We restarted a mission...
+         not (ca.getNextMissionFilename() == nil)) --...and is not the end of the campaign.
+        ) then
+            self.LSPipe:write "split\n" --Split time.
+            self.LSPipe:flush()
+        end
+        self.LSPipe:write "pausegametime\n" --Stop the clock for the loading screen.
+        self.LSPipe:flush()
+        self.CurrentMission = mn.getMissionFilename()
+    end
+
+    function SpeedrunTimer:StartTimer()
+        if(ca == nil) then
+            return
+        end
+        self.RunningStatus = true
+        self.LastStart = time.getCurrentTime()
+        ba.println("Speedrun Timer: StartTimer called")
+        self:StartLiveSplit()
+    end
+    
+    function SpeedrunTimer:StartLiveSplit() --Send Resume to LiveSplit
+        if not self.Started then
+            ba.println("Speedrun Timer: Run started, calling LiveSplit.")
+            self.LSPipe:write "start\n" --Start the game!
+            self.LSPipe:flush()
+            self.Started = true
+        else
+            ba.println("Speedrun Timer: Segment Unpaused, calling LiveSplit.")
+            self.LSPipe:write "unpausegametime\n" --Loading screen is done, unpause the clock.
+            self.LSPipe:flush()
+        end
+    end
+
+    function SpeedrunTimer:ResetLiveSplit()
+        if self.LSPipe == nil then
+            ba.warning("Speedrun Timer: Reset called, but the pipe does not exist.")
+        else
+            self.LSPipe:write "reset\n"
+            self.LSPipe:flush()
+            self.LSPipe:close()
+        end
+    end
+]
+
+$On Load Screen:
+[
+    if SpeedrunTimer.RunningStatus == true then
+        SpeedrunTimer:StopTimer()
+        ba.println("Speedrun Timer: Paused, Loading Screen")
+    end
+]
+
+$On Load Complete:
+[
+    if (mn == nil) then
```

Comment:
"mn" is also a library, so I don't think it is nil either.  You might want `mn.isInMission()`